streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -510,6 +526,8 @@ public void shouldReturnFalseOnCloseWhenThreadsHaventTerminated() throws Excepti
 
             assertTrue("Timed out waiting to receive single message", latch.await(30, TimeUnit.SECONDS));
             assertFalse(streams.close(Duration.ofMillis(10)));
+            // assertTrue(keyTestSerde.configured);
```

Comment:
Not sure atm why this does not pass -- the test setup seems correct to me. Try to see how the key and value serdes are passed along. I don't know this from the top of my head who the detailed flow is.

Reply:
Fixed the assertion in latest commit.

Reply:
One test should test only one thing IMHO. To share code, maybe refactor one private method that two public`@Test` methods can both use?